cpp/test/sg/genetic/evolution_test.cu

```diff
@@ -0,0 +1,327 @@
+/*
+ * Copyright (c) 2020-2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuml/genetic/common.h>
+#include <cuml/genetic/genetic.h>
+#include <cuml/genetic/node.h>
+#include <cuml/genetic/program.h>
+#include <gtest/gtest.h>
+#include <test_utils.h>
+#include <algorithm>
+#include <cmath>
+#include <cuml/common/logger.hpp>
+#include <iostream>
+#include <raft/handle.hpp>
+#include <rmm/device_uvector.hpp>
+#include <vector>
+
+namespace cuml {
+namespace genetic {
+
+/**
+ * @brief Tests the training and inference of the symbolic regressor, classifier and transformer
+ *        on y = 0.5X[0] + 0.4 X[1]
+ *
+ */
+class GeneticEvolutionTest : public ::testing::Test {
+ protected:
+  void SetUp() override
+  {
+    ML::Logger::get().setLevel(CUML_LEVEL_DEBUG);
+    CUDA_CHECK(cudaStreamCreate(&stream));
+    handle.set_stream(stream);
+
+    // Set training param vals
+    hyper_params.population_size       = 5000;
+    hyper_params.num_features          = n_cols;
+    hyper_params.random_state          = 11;
+    hyper_params.generations           = 20;
+    hyper_params.stopping_criteria     = 0.01;
+    hyper_params.p_crossover           = 0.7;
+    hyper_params.p_subtree_mutation    = 0.1;
+    hyper_params.p_hoist_mutation      = 0.05;
+    hyper_params.p_point_mutation      = 0.1;
+    hyper_params.parsimony_coefficient = 0.01;
+    // hyper_params.metric = metric_t::mae;
```

Comment:
Is this something we are planning to use in a follow-on or can it be removed?

Reply:
for now, have removed the parameter, but have kept it as a member so in the future we can add more metrics corey